lib/mongoid/persistable/datable.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Mongoid
+  module Persistable
+
+    # Defines behavior for setting a field (or fields) to the current
+    # date/time.
+    module Datable
+      extend ActiveSupport::Concern
+
+      # Converts the argument into the appropriate $currentDate type.
+      #
+      # @param [ nil | String | Symbol | Hash ] type The type description
+      #   to translate.
+      #
+      # @return [ true | Hash ] The type description indicated by the argument.
+      #
+      # @api private
+      def self.translate_date_field_spec(type)
+        case type
+        # `current_date(:foo)` == `current_date(foo: true)`
+        when nil then true
+        when "timestamp", :timestamp, "date", :date
+          { "$type" => type.to_s }
+        else type
```

Comment:
Shouldn't this be `else true`?

Reply:
Good eye, but no, this is intentional. The idea is that we can allow programmers to pass either `true` explicitly, or a Hash object representing the payload to send to the `$currentDate` command, e.g.:

```ruby
  model.current_date(stamp: { "$type" => "date" })
```

This is primarily to grant access to the underlying API, where (e.g.) `stamp: :date` is shorthand for `stamp: { "$type" => "date" }`. It is conceivable that a future version of MongoDB could expose new parameters here, and this would allow access to those even before Mongoid itself is updated.

Reply:
how about `when Hash` then instead? I can see people doing something like `model.current_date(stamp: true)`, `model.current_date(stamp: 1)`, etc.